compass/ocean/tests/nonhydro/stratified_seiche/initial_state.py

```diff
@@ -0,0 +1,221 @@
+            refZMid[k] = -refBottomDepth[k - 1] - 0.5 * refLayerThickness[k]
+
+        # SSH
+        ssh[:] = 0.0
+
+        # Compute maxLevelCell and layerThickness for z-level
+        # (variation only on top)
+        if (vertical_coordinate == 'z'):
+            vertCoordMovementWeights[:] = 0.0
+            vertCoordMovementWeights[0] = 1.0
+            for iCell in range(0, nCells):
+                maxLevelCell[iCell] = nVertLevels - 1
+                bottomDepth[iCell] = refBottomDepth[nVertLevels - 1] 
+                layerThickness[0, iCell, :] = refLayerThickness[:]
+                layerThickness[0, iCell, 0] += ssh[iCell]
+            restingThickness[:, :] = layerThickness[0, :, :]
+        # Compute maxLevelCell and layerThickness for uniform
+        elif (vertical_coordinate == 'uniform'):
+            vertCoordMovementWeights[:] = 1.0
+            vertCoordMovementWeights[0] = 1.0
+            for iCell in range(0, nCells):
+                maxLevelCell[iCell] = nVertLevels - 1
+                bottomDepth[iCell] = refBottomDepth[nVertLevels - 1]
+                layerThickness[0, iCell, :] = refLayerThickness[:] + \
+                    ssh[iCell]/nVertLevels
+            restingThickness[:, :] = refLayerThickness[:]
+
+        # Compute zMid (same, regardless of vertical coordinate)
+        for iCell in range(0, nCells):
+            k = maxLevelCell[iCell]
+            zMid[0, iCell, k] = -bottomDepth[iCell] + \
+                0.5 * layerThickness[0, iCell, k]
+            for k in range(maxLevelCell[iCell] - 1, -1, -1):
+                zMid[0, iCell, k] = zMid[0, iCell, k + 1] + 0.5 * \
+                    (layerThickness[0, iCell, k + 1] + \
+                    layerThickness[0, iCell, k])
+
+        # initialize tracers
+        rho0 = 1000.0  # kg/m^3
+        rhoz = -2.0e-4  # kg/m^3/m in z
+        S0 = 35.0
+        r = 1
+
+        # linear equation of state
+        # rho = rho0 - alpha*(T-Tref) + beta*(S-Sref)
+        # set S=Sref
+        # T = Tref - (rho - rhoRef)/alpha
+        for k in range(0, nVertLevels):
+            activeCells = k <= maxLevelCell
+            salinity[0, activeCells, k] = S0 
+            density[0, activeCells, k] = rho0 + (10.0/2)*(1.0 - \
+                np.tanh((2/1)*np.arctanh(0.99)*(zMid[0, :, k] + \
+                0.5*maxDepth - 0.1*np.cos((np.pi/10)*xCell[:]))))
+            # T = Tref - (rho - rhoRef)/alpha
+            temperature[0, activeCells, k] = config_eos_linear_Tref \
+                - (density[0, activeCells, k] - config_eos_linear_densityref) / \
+                config_eos_linear_alpha
+
+        # initial velocity on edges
+        ds['normalVelocity'] = (('Time', 'nEdges', 'nVertLevels',),
+            np.zeros([1, nEdges, nVertLevels]))
+        normalVelocity = ds['normalVelocity']
+        for iEdge in range(0, nEdges):
+            normalVelocity[0, iEdge, :] = u[0, iEdge, :] * math.cos(angleEdge[iEdge])
```

Comment:
This seems complicated, given that `u` is just all zeros (unless I missed something).  I think the following would do it:
```suggestion
        # initial velocity on edges
        ds['normalVelocity'] = (('Time', 'nEdges', 'nVertLevels',),
                                np.zeros([1, nEdges, nVertLevels]))
```
Do you have in mind to support non-zero `u` in the future?

Reply:
The lines
```
        normalVelocity = ds['normalVelocity']
        for iEdge in range(0, nEdges):
            normalVelocity[0, iEdge, :] = u[0, iEdge, :] * math.cos(angleEdge[iEdge])
```
are indeed there for the more general case of a non-zero initial velocity. So, in that case, all you have to do is changing `u`.

Reply:
Okay, that sounds reasonable.